hw_2/hw_2.py

```diff
@@ -0,0 +1,75 @@
+# Task 2 Part 1
+# Write a code, which will:
+# 1. create a list of random number of dicts (from 2 to 10)
+# dict's random numbers of keys should be letter,
+# dict's values should be a number (0-100),
+# example: [{'a': 5, 'b': 7, 'g': 11}, {'a': 3, 'c': 35, 'g': 42}]
+
+import random
+import string
+
+list_of_dicts = []
+number_of_dicts = random.randint(3, 10)
+
+
+def dict_sort(my_dict: dict) -> dict:  # function that gets dict as argument and return sorted dict
+    my_keys = list(my_dict.keys())  # getting list of keys for given dict
+    my_keys.sort()  # sorting list of keys from the dict
+    return {i: my_dict[i] for i in my_keys}  # returning dict with sorted keys
+
+
+# 'while' loop was selected because of by using 'for' loop may cause loss duplicates of keys f.e. if generated value for
+# amount of keys was 26, and we will have duplicates in keys value (2 times 'a' char) - we will have fewer keys in total
+while True:  # Loop for filling in list by dicts
+    this_dict = {}  # initialization temp dict variable to collect generated values
+    number_of_pairs_in_dict = random.randint(3, 26)  # initialization and generating value for number of pairs in dict
+    while True:  # Loop for filling in the dict
+        this_dict[''.join(random.choices(string.ascii_lowercase))] = random.randint(0, 100)  # adding random char as
+        # a key and random int in range 0-100 as a pair into the dict
+        if len(this_dict) == number_of_pairs_in_dict:  # Check if required amount of keys are already generated
+            break  # Closing the loop filling the dict
+    this_dict = dict_sort(this_dict)  # Sorting generated dict by function dict_sort
+    list_of_dicts.append(this_dict)  # Adding generated dict to the list of dicts
+
+    if len(list_of_dicts) >= number_of_dicts:  # Check length of list with dicts is less than required
+        break  # break for loop when list will be filled up appropriate count times
+
+# Task 2 Part 2
+# 2. get previously generated list of dicts and create one common dict:
+# if dicts have same key, we will take max value, and rename key with dict number with max value
+# if key is only in one dict - take it as is,
+# example:{'a_1': 5, 'b': 7, 'c': 35, 'g_2': 42}
+# Each line of code should be commented with description.
+
+new_dict = {}  # Initialization new dict variable
+dict_for_changed_elements = {}  # Initialization dict variable for changed elements (key = key, value = dict number)
+for x in range(len(list_of_dicts)):  # Loop for detecting duplicates in keys
```

Comment:
Better iterate with enumerate(list_of_dicts), you get 'currect_dict' and be able iterate through dictionary keys without using indexing 

```
for dict_num, dict in enumerate(list_of_dicts, start=1):   
    for key in dict:
        if key in new_dict:
```